code/game/objects/items/tools/maintenance_tools.dm

```diff
@@ -463,3 +463,68 @@
 	icon_state = "marine_flamerpack"
 	w_class = WEIGHT_CLASS_BULKY
 	max_fuel = 500 //Because the marine backpack can carry 260, and still allows you to take items, there should be a reason to still use this one.
+
+/obj/item/tool/handheld_charger
+	name = "Handheld charger"
+	desc = "A hand-held, lightweight cell charger. It isn't going to give you tons of power, but it can help in a pinch."
+	icon = 'icons/obj/items/items.dmi'
+	icon_state = "handheldcharger"
+	item_state = "handheldcharger"
+	w_class = WEIGHT_CLASS_SMALL
+	flags_atom = CONDUCT
+	force = 6.0
+	throw_speed = 2
+	throw_range = 9
+	flags_equip_slot = ITEM_SLOT_BELT
+	materials = list(/datum/material/metal = 50, /datum/material/glass = 20)
+	/// This is the cell we ar charging
+	var/obj/item/cell/cell
+
+/obj/item/tool/handheld_charger/attack_self(mob/user)
+	if(!cell)
+		to_chat(src, "You need some cell to be useful, idiot")
+		return
+	if(!do_after(user, 5, TRUE, src, BUSY_ICON_CLOCK))
+		return
```

Comment:
0.5 SECONDS

Also you can make this loop with a while loop if you dont want people to spam attack self

Reply:
Implemented